### PR TITLE
WIP: Combined contour function with slicer to use affine

### PR DESCRIPTION
**Pull request by kesshijordan**
I took the code from the contour function from fvtk.py (which doesn't use an affine) and combined it with code from the slicer in actor.py to make a function contour_actor in actor.py that uses an affine to place contours. I'm using this to display a roi with background image (slicer) and streamlines. I haven't written tests, yet.

### CODE REVIEW

File: dipy/viz/actor.py [hunk 1]
```diff
@@ -201,6 +201,139 @@ def copy(self):
     return image_actor
 
 
+def contour_actor(data, affine=None, levels=[50],
+            colors=[np.array([1.0, 0.0, 0.0])], opacities=[1]):
```
Comment (arokem) [inline dipy/viz/actor.py]: PEP8: indentation level here doesn't match the line above (`colors` needs to be flush with `data`)

File: dipy/viz/actor.py [hunk 2]
```diff
@@ -201,6 +201,139 @@ def copy(self):
...
+        skin_actor.GetProperty().SetOpacity(opacities[i])
+
+        skin_actor.GetProperty().SetColor(colors[i][0], colors[i][1], colors[i][2])
+        skin_assembly.AddPart(skin_actor)
+        del skin_actor
+        del skinMapper
+        del skinExtractor
+
+    return skin_assembly
+
```
Comment (arokem) [inline dipy/viz/actor.py]: One more line of white-space between functions please.

File: dipy/viz/actor.py (hunk 1 above)
Comment (kesshijordan) [inline dipy/viz/actor.py]: Thanks, ariel! I just added PEP8 comments to my pycharm so this should happen less. It doesn't like the default values I set; is this fine for dipy or is there another way I should do this? (It hilights levels, colors, and opacities with this comment: "This inspection detects when a mutable value as list or dictionary is detected in a default value for an argument. Default argument values are evaluated only once at function definition time, which means that modifying the default value of the argument will affect all subsequent calls of the function. ") Thanks! -Kesshi

Comment (kesshijordan) [inline dipy/viz/actor.py]: (Fixed this)

File: dipy/viz/actor.py (hunk 2 above)
Comment (kesshijordan) [inline dipy/viz/actor.py]: Fixed this

**Comment by kesshijordan** [comment]
Thanks, Ariel!

Yes; you can add it as an actor. I used the same code as was already implemented in the fvtk.py contour function, but I combined it with the slicer code so that it takes an affine (when I used the contour function originally, it rendered successfully, but was placed in the wrong part of the brain). The previous version took an affine, but didn't use it. See, below, how it's used for a binary roi.

-Kesshi

roidata = roidata.astype('uint8')
roi_actor = actor.contour_actor(roidata, levels=[1], affine=affine)
fvtk.add(ren,roi_actor)

**Comment by arokem** [comment]
Hey @kesshijordan - did you mean to push the bootstrap_dg stuff into this same PR?

**Comment by kesshijordan** [comment]
I just messed this up with git and merged with something that isn't ready.

**Comment by arokem** [comment]
(it should be a separate PR, in my opinion)

**Comment by arokem** [comment]
Gotcha. Let me know if you need any help rebasing this out of here.

**Comment by kesshijordan** [comment]
Yes; I'm trying to fix it. Didn't mean to merge that.

**Comment by kesshijordan** [comment]
Thanks, @arokem. I'm just going to make it a new PR and close this one.

**Pull request closed without merging**